onedal/tests/test_common.py

```diff
@@ -14,27 +14,43 @@
 # limitations under the License.
 # ==============================================================================
 
+import importlib
 import os
 from glob import glob
 
-import pytest
 
-
-def test_sklearn_check_version_ban():
-    """This test blocks the use of sklearn_check_version
-    in onedal files. The versioning should occur in the
-    sklearnex package for clarity and maintainability.
+def _test_primitive_usage_ban(primimtive_name, banned_locations, allowed_locations=None):
+    """This test blocks the usage of the primitive in
+    in certain files.
     """
-    from onedal import __file__ as loc
+
+    loc = importlib.import_module(banned_locations).__file__
```

Comment:
```suggestion
    loc = pkgutil.get_loader(package).filename
```
This removes the need to import the package, probably something that I should have done in the original implementation.